### PR TITLE
ci(actions): migrate to file-based commands

### PR DESCRIPTION
**Pull request by neersighted**
**- What I did**
Use the new `$GITHUB_STATE` and `$GITHUB_OUTPUT` file-based command systems instead of the deprecated commands on stdout.

Likewise, bump versions in Linux actions CI to pick up new releases of actions with the same fix.

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

### CODE REVIEW

File: .github/workflows/.windows.yml [hunk 1]
```diff
@@ -231,7 +231,7 @@ jobs:
           # Also prepend ./... to the matrix. This is a special case to run "Test integration" step exclusively.
           matrix="$(gotestlist -d ${{ env.ITG_CLI_MATRIX_SIZE }} ./...)"
           matrix="$(echo "$matrix" | jq -c '. |= ["./..."] + .')"
-          echo "::set-output name=matrix::$matrix"
+          echo "matrix=$matrix" >> $GITHUB_STATE
```
Comment (crazy-max) [inline .github/workflows/.windows.yml:234]: Looks it breaks the matrix integration tests on windows. Looks empty now: https://github.com/moby/moby/actions/runs/3439139750/jobs/5736096920

Comment (crazy-max) [reply .github/workflows/.windows.yml:234]: Ah it should be:

```suggestion
          echo "matrix=$matrix" >> $GITHUB_OUTPUT
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -76,7 +76,7 @@ jobs:
         id: scripts
         run: |
           scripts=$(jq -ncR '[inputs]' <<< "$(ls -I .validate -I all -I default -I dco -I golangci-lint.yml -I yamllint.yaml -A ./hack/validate/)")
-          echo "::set-output name=matrix::$scripts"
+          echo "matrix=$scripts" >> $GITHUB_STATE
```
Comment (crazy-max) [inline .github/workflows/test.yml:79]:
```suggestion
          echo "matrix=$scripts" >> $GITHUB_OUTPUT
```

**Review by crazy-max — APPROVED** [review]
LGTM thx!

**Comment by crazy-max** [comment]
Do we close https://github.com/moby/moby/pull/44177 btw?

**Comment by neersighted** [comment]
> Do we close #44177 btw?

I think we should still get that one in, rebased on top of this PR. I did not bump `upload-artifact` as it's not really in-scope for this PR, and your addition of dependabot will help manage these issues in the future.

**Review by crazy-max — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Comment by thaJeztah** [comment]
@neersighted can you prepare a 🍒 ⛏️ ?

**Pull request merged**